ivy/functional/frontends/jax/numpy/mathematical_functions.py

```diff
@@ -624,6 +624,31 @@ def polyint(p, m=1, k=None):
     return ivy.divide(ivy.concat((p, k_arr)), coeff).astype(p.dtype)
 
 
+@with_unsupported_dtypes(
+    {"0.3.14 and below": ("float16",)},
+    "jax",
+)
+@to_ivy_arrays_and_back
+def polydiv(u, v, *, trim_leading_zeros=False):
+    dtypeu = u.dtype
+    dtypev = v.dtype
+    u = ivy.asarray(u)
+    v = ivy.asarray(v)
```

Comment:
I see in the CI checks there is an error caused by these `ivy.asarray` calls. Could you explain why you think they are needed? `u` and `v` are already converted to ivy arrays in `@to_ivy_arrays_and_back`.

Reply:
I added those to fix an issue with tensorflow. It was returning two different dtypes but I removed them and realized it was my mistake. I'll check logic but as far as I remember, just followed the original. Thanks for explanations.